### DISCUSSION TITLE
unable to controll pagination component

### DISCUSSION THREAD

**Discussion by AbyerAli** [mantinedev/mantine#4038] [Q&A]
```
const [activePage, setPage] = useState(1);
<Pagination`  value={activePage} onChange={setPage} total={Math.ceil(allProductsCount / 10) || 1} color="dark" size="sm" />

```
this is not working, apparently the state is changing and i can see the results but when i rerender the pagination component specifically: even though the state is maintained but the pagination component resets to initial value i.e. 1.
any help please.

**Comment by rtivital** [comment]
Provide a sandbox with minimal reproduction, template – https://codesandbox.io/s/mantine-template-9ze89?file=/src/App.tsx:0-6

**Reply by AbyerAli** [reply]
I've updaed the sandbox,
its working just fine on the sandbox though, but i've tested the same thing on my app with this same minimal code and its not working
on changing the key the from UI prespective the page component resets to first page even though the state is not 1 (it may be 2 or any other value).

**Reply by AbyerAli** [reply]
![image](https://user-images.githubusercontent.com/59156204/231112283-416ecd4e-5240-4273-8a05-90dc625559ab.png)
this is the state i see from dev tools

![image](https://user-images.githubusercontent.com/59156204/231113136-4cf7fde4-8fb4-44fe-a3be-be00f385499f.png)
and this is the UI at that time.
